os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -488,6 +512,17 @@ int8_t cmd_wifi_connect(trwifi_ap_config_s *ap_connect_config, void *arg)
 		return -1;
 	}
 
+	if (rtw_memcmp(ssid, &wifi_ap_info.ssid, strlen(ssid))) {
+		channel = wifi_ap_info.channel;

```

Comment:
If connected AP changes channel later(SSID is same), then does it connect to the AP? Please check mentioned case.

Reply:
> If connected AP changes channel later(SSID is same), then does it connect to the AP? Please check mentioned case.

@pillip8282 , If the channel is modified after leave and tries to connect to same AP, there will be connect fail for the 1st connect after channel modification with the implementation. The pscan will start the scan with previous channel, it will fail first but we memset "wifi_ap_info" after every pscan irrespective of success/fail. 
Since it is a fail case, the profile will be empty inside "wlan_store_connect_info()". For the second round, it does full scan and connects normally & stores the latest profile (same SSID and modified channel information) for the subsequent rounds.
We asked Eric to check the behavior of TP2x for the same case and he confirmed that it will also have one connect fail after channel modification.

Reply:
Then upper layer(mostly wifi_manager) must call trwifi_connect_ap() twice in that case?
1) tp1x leaves an AP.
2) AP change its channel
3) tp1x connect to an AP via trwifi_connect_ap()
4) A wi-fi driver returns error
5) wifi_manager calls trwifi_connect_ap() again
6) A wi-fi driver returns success.
Is above sequence right?
If so, how about to modify trwifi_connect_ap() logic even though tp2x does same.
If pscan failed with previous channel, how about to do full scan not to return errors to wifi_manager immediately.

Reply:
> Then upper layer(mostly wifi_manager) must call trwifi_connect_ap() twice in that case?
> 
> 1. tp1x leaves an AP.
> 2. AP change its channel
> 3. tp1x connect to an AP via trwifi_connect_ap()
> 4. A wi-fi driver returns error
> 5. wifi_manager calls trwifi_connect_ap() again
> 6. A wi-fi driver returns success.
> 
> Is above sequence right?
> If so, how about to modify trwifi_connect_ap() logic even though tp2x does same.
> If pscan failed with previous channel, how about to do full scan not to return errors to wifi_manager immediately.
> To reduce time, I think suggested logic will be fast.

@pillip8282, do you suggest  "5. wifi_manager calls trwifi_connect_ap() again" in case of channel modification scenario? 
The current implementation is,
1.tp1x leaves an AP.
2.AP change its channel
3.tp1x connect to an AP via trwifi_connect_ap()
4.A wi-fi driver returns error which in turn returns error to wifi_manager 

Proposed logic:
1.tp1x leaves an AP.
2.AP change its channel
3.tp1x connect to an AP via trwifi_connect_ap()
4.A wi-fi driver returns error
5.wifi_manager calls trwifi_connect_ap() again
6.A wi-fi driver returns success which in turn returns success to wifi_manager

Is that correct?